gamemodes/terrortown/gamemode/client/cl_equip.lua

```diff
@@ -1115,17 +1115,23 @@ end
 -- @param table item
 -- @realm client
 function TTT2CacheEquipMaterials(item)
-	--if there is no material or model, the item should probably not be available in the shop
-	if item.material and item.material ~= "vgui/ttt/icon_id" then
-		item.ttt2_cached_material = Material(item.material)
-		if item.ttt2_cached_material:IsError() then
-			-- Setting fallback material
-			item.ttt2_cached_material = fallback_mat
+	item.isEquipment = true
+
+	if item.material then
+		item.iconMaterial = Material(item.material)
+
+		if item.iconMaterial:IsError() then
+			-- setting fallback error material
+			item.iconMaterial = fallback_mat
 		end
-	elseif item.model and item.model ~= "models/weapons/w_bugbait.mdl" then
-		--do not use fallback mat and use model instead
-		item.ttt2_cached_material = nil
-		item.ttt2_cached_model = item.model
+	elseif item.model then
+		-- do not use fallback mat and use model instead
+		item.itemModel = item.model
+	end
+
+	--if there is no sensible material or model, the item should probably not be available in the shop
+	if item.material == "vgui/ttt/icon_id" or item.model == "models/weapons/w_bugbait.mdl" then
+		item.isEquipment = false
```

Comment:
This Change probably deactivates equipment editor. Im investigating it further and see if I can fix it.

Reply:
Are you sure about this? Because the equipment editor worked fine in our last testsession. And this was already in there at that point

Reply:
Not about this code part, it seems fine, but this PR definitively makes the editor unusable at my end. Master works.

Reply:
I see now why. The weapons all have "models/weapons/w_bugbait.mdl" and are therefore marked as false equipment.
Before this change you could use them as long as they had a material.

Reply:
sounds like something broke on one of the merges or review requests

Reply:
> I see now why. The weapons all have "models/weapons/w_bugbait.mdl" and are therefore marked as false equipment. Before this change you could use them as long as they had a material.

ah, then probably the `or` has to be changed to an `and`

Reply:
Well also no, as for the equipment editor you want a different logic than the shop. In the shop there should only be weapons with icons and the editor should be able to handle all.

But that logic was scuffed anyway. So I'll keep looking into it.

Reply:
ah, so I replaced the bad hack with an even worse hack, I see. Maybe we merge it like it is right now and then replace it immediately with a new PR that fixes this in a good way?

Reply:
Did you need that change here in the PR? Otherwise I's say reverse it and we do it another time.

Reply:
> > I see now why. The weapons all have "models/weapons/w_bugbait.mdl" and are therefore marked as false equipment. Before this change you could use them as long as they had a material.
> 
> ah, then probably the `or` has to be changed to an `and`

My bad, you are right. Apparently all this caching was only used in our editors. So changing it to and fixes it. Im uploading the fix.